src/main/java/com/spotify/folsom/client/Request.java

```diff
@@ -65,4 +72,32 @@ protected static ByteBuf toBuffer(final ByteBufAllocator alloc, ByteBuffer dst,
   public int getOpaque() {
     return opaque;
   }
+
+  @VisibleForTesting
+  static byte[] encodeKey(String key, Charset charset) {
+    checkNotNull(key, "key");
```

Comment:
Sounds like a good idea (though it can never be null if it's used through the MemcacheClientBuilder class)